Code/Framework/AzCore/AzCore/Serialization/SerializeContext.h

```diff
@@ -2558,15 +2558,5 @@ namespace AZ
     template <typename, bool, bool> \
     friend struct AZ::Serialize::InstanceFactory;
 
-/// include AZStd containers generics
-#include <AzCore/Serialization/AZStdContainers.inl>
-#include <AzCore/Serialization/std/VariantReflection.inl>
-
-/// include asset generics
-#include <AzCore/Asset/AssetSerializer.h>
-
-/// include implementation of SerializeContext::EnumBuilder
-#include <AzCore/Serialization/SerializeContextEnum.inl>
-
```

Comment:
In a follow up pass we can reduce the number includes even more, if we move out the inner classes from outside the SerializeContext itself, to allow for more forward declarations.
i.e `SerializeContext::DataElement -> SerializeContextDataElement`
To reduce the blast radius of what has to change, the DataElement identifier can be become a type alias inside of the SerializeContext

i.e
### SerializeContextDataElement.h 
```
class SerializeContextDataElement
{
 // ...
};
```

### SerializeContext.h
```
class SerializeContext
{
    using DataElement = SerializeContextDataElement;
};
```

Reply:
I have a change in progress for that direction on `cmake/SPEC-7180`